### PR TITLE
fixes #24742 : Added test for adaptiveThreshold()

### LINKED ISSUE #24742: adaptiveThreshold is not covered with accuracy tests
### System Information

Any OS

### Detailed description

There is no C++ accuracy test for the mentioned function.

### Steps to reproduce

-

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

### PR DESCRIPTION
**Pull request by Dhanwanth1803**
@dkurt Sir I added `test_adaptiveThereshold( const Mat& _src, Mat& _dst, double maxValue, int method, int type, int blockSize, double delta)` like you said. I made builds pass locally. Can you please take a look.

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [X] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/imgproc/test/test_thresh.cpp [hunk 1]
```diff
@@ -478,6 +478,221 @@ static void test_threshold( const Mat& _src, Mat& _dst,
     }
 }
 
+//test for adaptive Threshold
+static void test_adaptiveThereshold( const Mat& _src, Mat& _dst, double maxValue,
```
Comment (dkurt) [inline modules/imgproc/test/test_thresh.cpp:482]: I don't see any invocation of `cv::adaptiveThreshold` here.

**Review by dkurt — CHANGES_REQUESTED** [review]

File: modules/imgproc/test/test_thresh.cpp (hunk 1 above)
Comment (Dhanwanth1803) [reply modules/imgproc/test/test_thresh.cpp:482]: Sir, you only told to start writing test function in `test_thresh` and then open a PR.

Comment (Dhanwanth1803) [reply modules/imgproc/test/test_thresh.cpp:482]: Plus, @dkurt  you to told me told `cv::threshold` as reference. Neither `test_threshold` has an invocation of `cv::threshold`

Comment (Dhanwanth1803) [reply modules/imgproc/test/test_thresh.cpp:482]: @dkurt . Sir, I don't understand where the values are being verified in the file `test_thresh.cpp`. As there is no code where the expected and actual values from the function are compared in the file. From the code it seems that we only initialized and grabbed the values but never actually compare them. Is there something that is going on internally or am I missing something.

**Comment by Dhanwanth1803** [comment]
@dkurt Can I continue to work on this or should I leave it?

**Comment by dkurt** [comment]
@Dhanwanth1803, I recommend to pause this PR. Please focus on other issues. Thanks.

**Pull request closed without merging**